src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/state/Task.kt

```diff
@@ -0,0 +1,184 @@
+
+enum class Task {
+
+  ASSESS_ELIGIBILITY {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.eligibilityChecksStatus) {
+      EligibilityStatus.NOT_STARTED -> READY_TO_START
+      EligibilityStatus.IN_PROGRESS -> IN_PROGRESS
+      else -> COMPLETE
+    }
+  },
+
+  ENTER_CURFEW_ADDRESS {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      NOT_STARTED, ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS, INELIGIBLE_OR_UNSUITABLE -> LOCKED
+      ELIGIBLE_AND_SUITABLE -> READY_TO_START
+      AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS -> COMPLETE
+      OPTED_OUT -> IN_PROGRESS
+      else -> null
+    }
+  },
+
+  REVIEW_APPLICATION_AND_SEND_FOR_DECISION {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      NOT_STARTED, ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS, ELIGIBLE_AND_SUITABLE, AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS, INELIGIBLE_OR_UNSUITABLE, OPTED_OUT -> LOCKED
+      AWAITING_PRE_DECISION_CHECKS -> READY_TO_START
+      else -> null
+    }
+  },
+
+  COMPLETE_14_DAY_CHECKS {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      NOT_STARTED, ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS, ELIGIBLE_AND_SUITABLE, AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS, INELIGIBLE_OR_UNSUITABLE, OPTED_OUT -> LOCKED
+      else -> null
+    }
+  },
+
+  COMPLETE_2_DAY_CHECKS {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      NOT_STARTED, ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS, ELIGIBLE_AND_SUITABLE, AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS, INELIGIBLE_OR_UNSUITABLE, OPTED_OUT -> LOCKED
+      else -> null
+    }
+  },
+
+  PRINT_LICENCE {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      NOT_STARTED, ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS, ELIGIBLE_AND_SUITABLE, AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS, INELIGIBLE_OR_UNSUITABLE, OPTED_OUT -> LOCKED
+      else -> null
+    }
+  },
+
+  CONSULT_THE_VLO_AND_POM {
+    override fun visibleForRoles() = setOf(PROBATION_COM)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      AWAITING_ADDRESS_AND_RISK_CHECKS ->
+        if (assessment.victimContactSchemeOptedIn != null) COMPLETE else READY_TO_START
+
+      ADDRESS_AND_RISK_CHECKS_IN_PROGRESS -> COMPLETE
+      else -> null
```

Comment:
this could be simplified to: 
```kt
if (assessment.victimContactSchemeOptedIn != null) {
   return COMPLETE
} 
if (assessmentStatus == AWAITING_ADDRESS_AND_RISK_CHECKS){
   return NOT_STARTED
}
return null
```

Reply:
I like the when TBO